python/triton/compiler/code_generator.py

```diff
@@ -591,12 +591,12 @@ def visit_Assign(self, node):
         def _sanitize_value(value):
             if isinstance(value, language.tuple):
                 return _apply_to_tuple_values(value, _sanitize_value)
-            native_nontensor_types = (language.dtype, language.tuple)
-            value = _unwrap_if_constexpr(value)
+            native_nontensor_types = (language.dtype, language.tuple, int, float, bool)
```

Comment:
I'm confused about the change. If `x = 1.0`, shouldn't we make 1.0 a tensor first?

Reply:
Thanks for catching that.

The reason I wanted to make this change is that it causes `tuple[constexpr[int], ...]` (representing a shape) to be converted to a `tuple[tensor, ...]` (which is no longer a valid shape).

But, following your comment, I suppose that there's no way to differentiate between a tuple of ints representing a tuple of tensors, and a tuple of ints representing a shape.

Do you think that this current behavior (shapes being converted to `tuple[tensor, ...]`, but only when assigned to a variable) is something that should be changed or is it intended?

Reply:
I feel like the integers inside `self.shape = tuple([constexpr(s) for s in self.shape])` shouldn't be unwrapped

Reply:
> But, following your comment, I suppose that there's no way to differentiate between a tuple of ints representing a tuple of tensors, and a tuple of ints representing a shape.

I agree

Reply:
Maybe `tensor.shape` should be its own type instead of `tuple`?

Reply:
That workaround fixes this issue for me - thanks for the suggestion!